hazelcast/src/main/java/com/hazelcast/spi/impl/InternalOperationService.java

```diff
@@ -35,15 +34,13 @@
  */
 public interface InternalOperationService extends OperationService {
 
-    void onMemberLeft(MemberImpl member);
```

Comment:
It doesn't belong in the Interface. 

It depends on the implementation if and how he wants to listen to member left. If he wants to implement an interface or provide some internal class that does the listening, he should be able to do that any way he sees fit. It isn't part of the interface to enforce that. So you are mixing in an implementation detail into the interface. 

So if the BasicOperationService wants to be registered as a MemberListener, he could e.g. implement the MemberListener interface and register himself as listener to the cluster. This can be done in the BasicOperationService when it is constructed.